### PR TITLE
facebook login/signup

### PR DESCRIPTION
**Pull request by peter7z**
### Description

This PR adds functionality to the existing "connect with facebook" button.
It also establishes a skeleton for the home page.

### CODE REVIEW

File: src/actions/sessionActions.js [hunk 1]
```diff
@@ -5,14 +5,25 @@ import sessionApi from '../api/sessionApi';
 
 export const login = user =>
   () =>
-    sessionApi.login({ user }).then(({ user }) => {
+    sessionApi.login({ user }).then(({ data: user }) => {
```
Comment (fedeagripa) [inline src/actions/sessionActions.js:8]: using `{data}` does not work ? If you chosen this due to clarity if fine, just asking

**Review by fedeagripa — APPROVED** [review]

File: src/components/user/LoginForm.js [hunk 2]
```diff
@@ -60,9 +62,14 @@ export const LoginForm = ({ handleSubmit, error, submitting, intl }) => (
       <a className="forgot-password">
         <FormattedMessage id="login.forgot_password" />
       </a>
-      <a className="connect-facebook" >
-        <FormattedMessage id="login.facebook" />
-      </a>
+      <FacebookLogin
+        appId="166925907359293"
```
Comment (xlasserre) [inline src/components/user/LoginForm.js]: You could store this id in a constants file.

File: src/components/user/LoginForm.js [hunk 3]
```diff
@@ -78,7 +85,8 @@ LoginForm.propTypes = {
   handleSubmit: func.isRequired,
   intl: intlShape.isRequired,
   submitting: bool.isRequired,
-  error: string
+  error: string,
+  onFbLogin: func
```
Comment (xlasserre) [inline src/components/user/LoginForm.js]: Shouldn't this function be required?

**Review by xlasserre — CHANGES_REQUESTED** [review]
Good PR! 🙂I added just two comments and also wanted to ask you if you can share a screenshot to see how it looks like now. Thanks!

**Comment by peter7z** [comment]
@fedeagripa { data } does work. It's just for legibility.

File: src/actions/sessionActions.js [hunk 4]
```diff
@@ -5,14 +5,25 @@ import sessionApi from '../api/sessionApi';
...
       sessionService.saveUser(user);
     }).catch((err) => {
       throw new SubmissionError({
         _error: err.errors
       });
     });
 
+export const loginFacebook = user =>
+  () => {
+    sessionApi.login({ access_token: user }).then(({ data: user }) => {
```
Comment (felipegarcia92) [inline src/actions/sessionActions.js]: No need to use snake case. You just use camel case and then in `apiService.js` the keys get converted to snake via the `humps` package.

File: src/components/user/LoginForm.js [hunk 5]
```diff
@@ -2,13 +2,15 @@ import React from 'react';
...
 import { Link } from 'react-router-dom';
+import FacebookLogin from 'react-facebook-login';
 import {
   injectIntl,
   intlShape,
   defineMessages,
   FormattedMessage
 } from 'react-intl';
 
+import fbID from '../../constants/constants';
```
Comment (felipegarcia92) [inline src/components/user/LoginForm.js]: Names should match with the ones in the file.

Also when you're exporting more tha one thing you'll need the `{ ... }` notation to pull things out of it.

File: src/containers/LoginPage.js [hunk 6]
```diff
@@ -24,14 +24,16 @@ const LoginPage = ({ login, authenticated }) => {
...
 };
 
 const mapState = state => ({
   authenticated: state.getIn(['session', 'authenticated'])
 });
 
 const mapDispatch = dispatch => ({
-  login: user => dispatch(login(user.toJS()))
+  login: user => dispatch(login(user.toJS())),
+  loginFacebook: user => dispatch(loginFacebook(user.accessToken))
```
Comment (felipegarcia92) [inline src/containers/LoginPage.js]: If you just need the access token you can destructure the user parameter.

File: src/styles/styles.scss [hunk 7]
```diff
@@ -25,3 +26,16 @@ html {
   .loader-hidden { display: none; }
   .loader-active { display: block; }
 }
+
+.target-title {
+  color: $_BLACK;
```
Comment (felipegarcia92) [inline src/styles/styles.scss]: Can you rename it to just `$black`? That's how you name things in sass

**Review by felipegarcia92 — CHANGES_REQUESTED** [review]

**Comment by peter7z** [comment]
@missx No screenshots because styling and interface haven't changed. Home is still WIP, will leave for next PR.

**Review by xlasserre — APPROVED** [review]

**Pull request merged**